tests/phpunit/Unit/support/renderNewsletterSignupCallout.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ *  Tests for render_newsletter_signup_callout()
+ *
+ * @since      1.0.0
+ * @author     Robert A. Gadon
+ * @package    spiralWebDb\ExtendGiveWP\Tests\Unit
+ * @link       http://spiralwebdb.com
+ * @license    GNU General Public License 2.0+
+ */
+
+namespace spiralWebDb\ExtendGiveWP\Tests\Unit;
+
+use Brain\Monkey\Functions;
+use spiralWebDb\ExtendGiveWP\tests\phpunit\Unit\TestCase;
+use function spiralWebDb\ExtendGiveWP\render_newsletter_signup_callout;
+
+/**
+ * Class Test_RenderNewsletterSignupCallout
+ *
+ * @covers ::\spiralWebDb\ExtendGiveWP\render_newsletter_signup_callout
+ *
+ * @group   extend-give-wp
+ * @group   support
+ *
+ * phpcs:disable Squiz.Commenting.FunctionComment.MissingParamTag
+ */
+class Test_RenderNewsletterSignupCallout extends TestCase {
+
+	/**
+	 * Prepares the test environment before each test.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		require_once EXTEND_GIVE_WP_ROOT_DIR . '/src/support/load-assets.php';
+	}
+
+	/**
+	 * Test render_newsletter_signup_callout() should render donation levels label given $form_id.
+	 *
+	 * @dataProvider addTestData
+	 */
+	public function test_should_render_recurring_donation_option_label( $post_data, $expected ) {
+		Functions\expect( 'get_give_donation_form_id' )
+			->zeroOrMoreTimes()
+			->with( 'form_id' )
+			->andReturn( $post_data['form_id'] );
+		Functions\expect( '_get_plugin_dir' )->andReturn( EXTEND_GIVE_WP_ROOT_DIR );
+		Functions\when( 'esc_attr' )->justEcho( $post_data['form_id'] );
```

Comment:
`esc_attr() does not echo; rather it returns.

Check out how we do it here. You can add this to your test case to stub a bunch of different functions https://github.com/wp-media/phpunit/blob/master/Unit/TestCase.php#L63

Reply:
Just checked. [You have that method in the test case](https://github.com/rgadon107/extend-give-wp/blob/master/tests/phpunit/Unit/TestCase.php#L46). You have to call it in order for it work though. You can do that in the `setUp()` method.

```php
	public function setUp() {
		parent::setUp();

		$this-> setup_common_wp_stubs();
	}
```

Reply:
@hellofromtonya I appreciate the reference to the unit TestCase from WPMedia. The mocking function in my unit TestCase that stubs out WP's escaping functions is properly formatted (using WPMedia's file as a standard to check against). I even tried setting the property 

>protected static $mockCommonWpFunctionsInSetUp = false;

from WPMedia’s unit TestCase, and then setting a conditional in the `setUp()` method: 

		if ( static::$mockCommonWpFunctionsInSetUp ) {
			$this->mockCommonWpFunctions();
		}

Still didn’t work. The CLI returns the message: 

>Error: Call to undefined function esc_attr()

There’s something else about the setup of my test suite that’s off. Just can’t put my finger on it.